lib/rage/request.rb

```diff
@@ -14,6 +14,48 @@ def headers
     @headers ||= Headers.new(@env)
   end
 
+  # Check if the request is fresh.
+  # @param etag [String] The etag of the requested resource.
+  # @param last_modified [Time] The last modified time of the requested resource.
+  # @return [Boolean] True if the request is fresh, false otherwise.
+  # @example
+  #  request.fresh?(etag: "123", last_modified: Time.utc(2023, 12, 15))
+  #  request.fresh?(last_modified: Time.utc(2023, 12, 15))
+  #  request.fresh?(etag: "123")
+  def fresh?(etag:, last_modified:)
+    if_none_match = headers["HTTP_IF_NONE_MATCH"]
+    if_not_modified_since = headers["HTTP_IF_MODIFIED_SINCE"]
+
+    # Always render response when no freshness information
+    # is provided in the request.
+    return false unless if_none_match || if_not_modified_since
+
+    etag_matches?(
+      requested_etags: if_none_match, response_etag: etag
+    ) && not_modified?(
+      request_not_modified_since: if_not_modified_since,
+      response_last_modified: last_modified
+    )
+  end
+
+  private
+
+  def etag_matches?(requested_etags:, response_etag:)
+    requested_etags = requested_etags ? requested_etags.split(",").map(&:strip) : []
```

Comment:
```suggestion
    requested_etags = requested_etags ? requested_etags.split(",").map! { |tag| tag.strip! || tag } : []
```

Reply:
Also, do we have to create an empty array if there are no `requested_etags`? Can we go with `nil` instead?

Reply:
The thing here is that
<img width="352" alt="Screenshot 2023-12-22 at 10 25 06" src="https://github.com/rage-rb/rage/assets/3327396/705a5e2b-3633-4d8b-b4c6-19708aaa02d6">

which means that leaving it this way would mean that an empty string would be the same as not sending the header. Not sure if that's the behaviour we're looking for 🤔 
I will check how rails does it.

Also:
Using `each` instead of `map!` here is even faster and also makes more sense IMO, because we're already mutating the strings in place, so need to mutate the array at all.

Reply:
[ba6b43c](https://github.com/rage-rb/rage/pull/55/commits/ba6b43ccb214862ce47a4e7cbbf5fe7e74786f45)